portal/plugins/gnmatomresponder/master_importer.py

```diff
@@ -174,28 +174,62 @@ def register_pac_xml(self, content):
         record.save()
         return record
 
+    def check_for_old_finished_jobs(self, vs_item_id):
+        from models import ImportJob
+
+        jobs = ImportJob.objects.filter(item_id=vs_item_id).filter(status='FINISHED')
+
+        return len(jobs) > 0
+
+    def check_key(self, key, vs_item_id):
+        from models import ImportJob
+
+        jobs = ImportJob.objects.filter(item_id=vs_item_id).filter(s3_path=key)
+
+        return len(jobs) > 0
```

Comment:
doh! sorry.  I should have mentioned this before - there is a better way of doing this.  Put `.count()` onto the end of the `ImportJob.objects.filter` line and remove `len()` around `jobs` in the `return` statement.

This way we tell the _database itself_ to only return the count, rather than return all of the data and we count it ourselves.

Reply:
Good idea. I will try it.